# Cerial Prices Changes Within Last 30 Years
_____


In [90]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
%matplotlib inline

import plotly.express as px
from plotly.subplots import make_subplots
import plotly.graph_objects as go

import warnings
warnings.filterwarnings("ignore")

In [91]:
df = pd.read_csv('rice_wheat_corn_prices.csv')

In [92]:
df.head()

Year Month  Price_wheat_ton  Price_rice_ton  Price_corn_ton  \
0  1992   Feb           170.12          278.25          113.62   
1  1992   Mar           161.44          277.20          117.00   
2  1992   Apr           153.07          278.00          108.52   
3  1992   May           139.72          274.00          109.64   
4  1992   Jun           140.36          268.80          110.90   

   Inflation_rate  Price_wheat_ton_infl  Price_rice_ton_infl  \
0           89.59                322.53               527.53   
1           89.59                306.07               525.54   
2           89.59                290.21               527.06   
3           89.59                264.90               519.48   
4           89.59                266.11               509.62   

   Price_corn_ton_infl  
0               215.41  
1               221.82  
2               205.74  
3               207.87  
4               210.26

In [93]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 360 entries, 0 to 359
Data columns (total 9 columns):
 #   Column                Non-Null Count  Dtype  
---  ------                --------------  -----  
 0   Year                  360 non-null    int64  
 1   Month                 360 non-null    object 
 2   Price_wheat_ton       359 non-null    float64
 3   Price_rice_ton        360 non-null    float64
 4   Price_corn_ton        360 non-null    float64
 5   Inflation_rate        359 non-null    float64
 6   Price_wheat_ton_infl  359 non-null    float64
 7   Price_rice_ton_infl   359 non-null    float64
 8   Price_corn_ton_infl   359 non-null    float64
dtypes: float64(7), int64(1), object(1)
memory usage: 25.4+ KB


In [94]:
df.dtypes

Year                      int64
Month                    object
Price_wheat_ton         float64
Price_rice_ton          float64
Price_corn_ton          float64
Inflation_rate          float64
Price_wheat_ton_infl    float64
Price_rice_ton_infl     float64
Price_corn_ton_infl     float64
dtype: object

In [95]:
df.isnull().sum()

Year                    0
Month                   0
Price_wheat_ton         1
Price_rice_ton          0
Price_corn_ton          0
Inflation_rate          1
Price_wheat_ton_infl    1
Price_rice_ton_infl     1
Price_corn_ton_infl     1
dtype: int64

In [96]:
df.dropna(axis=0, inplace=True)

In [97]:
df.isnull().sum()

Year                    0
Month                   0
Price_wheat_ton         0
Price_rice_ton          0
Price_corn_ton          0
Inflation_rate          0
Price_wheat_ton_infl    0
Price_rice_ton_infl     0
Price_corn_ton_infl     0
dtype: int64

In [98]:
df.duplicated().sum()

0

In [99]:
df.head()

Year Month  Price_wheat_ton  Price_rice_ton  Price_corn_ton  \
0  1992   Feb           170.12          278.25          113.62   
1  1992   Mar           161.44          277.20          117.00   
2  1992   Apr           153.07          278.00          108.52   
3  1992   May           139.72          274.00          109.64   
4  1992   Jun           140.36          268.80          110.90   

   Inflation_rate  Price_wheat_ton_infl  Price_rice_ton_infl  \
0           89.59                322.53               527.53   
1           89.59                306.07               525.54   
2           89.59                290.21               527.06   
3           89.59                264.90               519.48   
4           89.59                266.11               509.62   

   Price_corn_ton_infl  
0               215.41  
1               221.82  
2               205.74  
3               207.87  
4               210.26

### Ploting

In [100]:
numerical_df = df.drop(columns='Month', axis=1)

In [101]:
fig = make_subplots(rows=1, cols=8)

n = 1
for column in numerical_df.columns:
    fig.add_trace(go.Box(y=numerical_df[column], name=column, boxpoints='suspectedoutliers'), row = 1, col = n)
    n += 1

for i in range(1,9):
    fig.update_xaxes(title_text=numerical_df.columns[i-1], tickangle=90, row=1, col=i)

fig.show()

In [102]:
df[df['Price_rice_ton_infl'] < 1120]

Year Month  Price_wheat_ton  Price_rice_ton  Price_corn_ton  \
0    1992   Feb           170.12          278.25          113.62   
1    1992   Mar           161.44          277.20          117.00   
2    1992   Apr           153.07          278.00          108.52   
3    1992   May           139.72          274.00          109.64   
4    1992   Jun           140.36          268.80          110.90   
..    ...   ...              ...             ...             ...   
354  2021   Aug           276.18          403.00          256.61   
355  2021   Sep           263.60          400.00          235.62   
356  2021   Oct           334.50          401.00          239.65   
357  2021   Nov           327.82          400.00          248.72   
358  2021   Dec           332.06          400.00          264.54   

     Inflation_rate  Price_wheat_ton_infl  Price_rice_ton_infl  \
0             89.59                322.53               527.53   
1             89.59                306.07               525.54   
2             89.59                290.21               527.06   
3             89.59                264.90               519.48   
4             89.59                266.11               509.62   
..              ...                   ...                  ...   
354           -1.29                272.62               397.80   
355           -1.29                260.20               394.84   
356           -1.29                330.18               395.83   
357           -1.29                323.59               394.84   
358           -1.29                327.78               394.84   

     Price_corn_ton_infl  
0                 215.41  
1                 221.82  
2                 205.74  
3                 207.87  
4                 210.26  
..                   ...  
354               253.30  
355               232.58  
356               236.56  
357               245.51  
358               261.13  

[358 rows x 9 columns]

# Data analysis

In [103]:
df.head()

Year Month  Price_wheat_ton  Price_rice_ton  Price_corn_ton  \
0  1992   Feb           170.12          278.25          113.62   
1  1992   Mar           161.44          277.20          117.00   
2  1992   Apr           153.07          278.00          108.52   
3  1992   May           139.72          274.00          109.64   
4  1992   Jun           140.36          268.80          110.90   

   Inflation_rate  Price_wheat_ton_infl  Price_rice_ton_infl  \
0           89.59                322.53               527.53   
1           89.59                306.07               525.54   
2           89.59                290.21               527.06   
3           89.59                264.90               519.48   
4           89.59                266.11               509.62   

   Price_corn_ton_infl  
0               215.41  
1               221.82  
2               205.74  
3               207.87  
4               210.26

We'll give month numbers

In [104]:
df.head()

Year Month  Price_wheat_ton  Price_rice_ton  Price_corn_ton  \
0  1992   Feb           170.12          278.25          113.62   
1  1992   Mar           161.44          277.20          117.00   
2  1992   Apr           153.07          278.00          108.52   
3  1992   May           139.72          274.00          109.64   
4  1992   Jun           140.36          268.80          110.90   

   Inflation_rate  Price_wheat_ton_infl  Price_rice_ton_infl  \
0           89.59                322.53               527.53   
1           89.59                306.07               525.54   
2           89.59                290.21               527.06   
3           89.59                264.90               519.48   
4           89.59                266.11               509.62   

   Price_corn_ton_infl  
0               215.41  
1               221.82  
2               205.74  
3               207.87  
4               210.26

We'll now categories them

In [105]:
df['Month_num'] = df['Month']

In [106]:
for x in df.index :
    if df.loc[x,'Month_num']=='Feb' :
        df.loc[x,'Month_num']=2
    elif df.loc[x,'Month_num']=='Mar' :
        df.loc[x,'Month_num']=3
    elif df.loc[x,'Month_num']=='Apr' :
        df.loc[x,'Month_num']=4
    elif df.loc[x,'Month_num']=='May' :
        df.loc[x,'Month_num']=5
    elif df.loc[x,'Month_num']=='Jun' :
        df.loc[x,'Month_num']=6
    elif df.loc[x,'Month_num']=='Jul' :
        df.loc[x,'Month_num']=7
    elif df.loc[x,'Month_num']=='Aug' :
        df.loc[x,'Month_num']=8
    elif df.loc[x,'Month_num']=='Sep' :
        df.loc[x,'Month_num']=9
    elif df.loc[x,'Month_num']=='Oct' :
        df.loc[x,'Month_num']=10
    elif df.loc[x,'Month_num']=='Nov' :
        df.loc[x,'Month_num']=11
    elif df.loc[x,'Month_num']=='Dec' :
        df.loc[x,'Month_num']=12
    elif df.loc[x,'Month_num']=='Jan' :
        df.loc[x,'Month_num']=1

In [107]:
df.head()

Year Month  Price_wheat_ton  Price_rice_ton  Price_corn_ton  \
0  1992   Feb           170.12          278.25          113.62   
1  1992   Mar           161.44          277.20          117.00   
2  1992   Apr           153.07          278.00          108.52   
3  1992   May           139.72          274.00          109.64   
4  1992   Jun           140.36          268.80          110.90   

   Inflation_rate  Price_wheat_ton_infl  Price_rice_ton_infl  \
0           89.59                322.53               527.53   
1           89.59                306.07               525.54   
2           89.59                290.21               527.06   
3           89.59                264.90               519.48   
4           89.59                266.11               509.62   

   Price_corn_ton_infl Month_num  
0               215.41         2  
1               221.82         3  
2               205.74         4  
3               207.87         5  
4               210.26         6

We'll check the average inflation

In [108]:
df[['Year', 'Inflation_rate']].groupby('Year').agg('mean')

Inflation_rate
Year                
1992           89.59
1993           84.15
1994           79.47
1995           74.58
1996           69.60
1997           65.73
1998           63.20
1999           59.70
2000           54.49
2001           50.24
2002           47.89
2003           44.61
2004           40.84
2005           36.22
2006           31.96
2007           28.30
2008           23.56
2009           24.00
2010           22.00
2011           18.27
2012           15.87
2013           14.19
2014           12.37
2015           12.24
2016           10.84
2017            8.53
2018            5.94
2019            4.06
2020            2.79
2021           -1.29

let's plot it

In [111]:
px.line(df[['Year', 'Inflation_rate']].groupby('Year').agg('mean'), markers='+')

In [113]:
df['Year']=df['Year'].astype(str)
df['Year']

0      1992
1      1992
2      1992
3      1992
4      1992
       ... 
354    2021
355    2021
356    2021
357    2021
358    2021
Name: Year, Length: 359, dtype: object

In [115]:
df['Date'] = df['Year'] + '-'+df['Month']
df['Date']

0      1992-Feb
1      1992-Mar
2      1992-Apr
3      1992-May
4      1992-Jun
         ...   
354    2021-Aug
355    2021-Sep
356    2021-Oct
357    2021-Nov
358    2021-Dec
Name: Date, Length: 359, dtype: object

In [117]:
df['Date'] = pd.to_datetime(df['Date'])
df['Date']

0     1992-02-01
1     1992-03-01
2     1992-04-01
3     1992-05-01
4     1992-06-01
         ...    
354   2021-08-01
355   2021-09-01
356   2021-10-01
357   2021-11-01
358   2021-12-01
Name: Date, Length: 359, dtype: datetime64[ns]

In [119]:
df[['Date', 'Inflation_rate']].groupby('Date').agg('mean')

Inflation_rate
Date                      
1992-02-01           89.59
1992-03-01           89.59
1992-04-01           89.59
1992-05-01           89.59
1992-06-01           89.59
...                    ...
2021-08-01           -1.29
2021-09-01           -1.29
2021-10-01           -1.29
2021-11-01           -1.29
2021-12-01           -1.29

[359 rows x 1 columns]

In [120]:
px.line(df[['Date', 'Inflation_rate']].groupby('Date').agg('mean'))

Inflation rate has decreased year on year except the the years 2008 and 2009 which has seen increase from 23.56 to 24

### Now lets see the inflation rate each from month from 1992

In [123]:
desc_df=df.pivot_table(index='Year',columns='Month_num',values='Inflation_rate',aggfunc='mean')
desc_df

Month_num     1      2      3      4      5      6      7      8      9   \
Year                                                                       
1992         NaN  89.59  89.59  89.59  89.59  89.59  89.59  89.59  89.59   
1993       84.15  84.15  84.15  84.15  84.15  84.15  84.15  84.15  84.15   
1994       79.47  79.47  79.47  79.47  79.47  79.47  79.47  79.47  79.47   
1995       74.58  74.58  74.58  74.58  74.58  74.58  74.58  74.58  74.58   
1996       69.60  69.60  69.60  69.60  69.60  69.60  69.60  69.60  69.60   
1997       65.73  65.73  65.73  65.73  65.73  65.73  65.73  65.73  65.73   
1998       63.20  63.20  63.20  63.20  63.20  63.20  63.20  63.20  63.20   
1999       59.70  59.70  59.70  59.70  59.70  59.70  59.70  59.70  59.70   
2000       54.49  54.49  54.49  54.49  54.49  54.49  54.49  54.49  54.49   
2001       50.24  50.24  50.24  50.24  50.24  50.24  50.24  50.24  50.24   
2002       47.89  47.89  47.89  47.89  47.89  47.89  47.89  47.89  47.89   
2003       44.61  44.61  44.61  44.61  44.61  44.61  44.61  44.61  44.61   
2004       40.84  40.84  40.84  40.84  40.84  40.84  40.84  40.84  40.84   
2005       36.22  36.22  36.22  36.22  36.22  36.22  36.22  36.22  36.22   
2006       31.96  31.96  31.96  31.96  31.96  31.96  31.96  31.96  31.96   
2007       28.30  28.30  28.30  28.30  28.30  28.30  28.30  28.30  28.30   
2008       23.56  23.56  23.56  23.56  23.56  23.56  23.56  23.56  23.56   
2009       24.00  24.00  24.00  24.00  24.00  24.00  24.00  24.00  24.00   
2010       22.00  22.00  22.00  22.00  22.00  22.00  22.00  22.00  22.00   
2011       18.27  18.27  18.27  18.27  18.27  18.27  18.27  18.27  18.27   
2012       15.87  15.87  15.87  15.87  15.87  15.87  15.87  15.87  15.87   
2013       14.19  14.19  14.19  14.19  14.19  14.19  14.19  14.19  14.19   
2014       12.37  12.37  12.37  12.37  12.37  12.37  12.37  12.37  12.37   
2015       12.24  12.24  12.24  12.24  12.24  12.24  12.24  12.24  12.24   
2016       10.84  10.84  10.84  10.84  10.84  10.84  10.84  10.84  10.84   
2017        8.53   8.53   8.53   8.53   8.53   8.53   8.53   8.53   8.53   
2018        5.94   5.94   5.94   5.94   5.94   5.94   5.94   5.94   5.94   
2019        4.06   4.06   4.06   4.06   4.06   4.06   4.06   4.06   4.06   
2020        2.79   2.79   2.79   2.79   2.79   2.79   2.79   2.79   2.79   
2021       -1.29  -1.29  -1.29  -1.29  -1.29  -1.29  -1.29  -1.29  -1.29   

Month_num     10     11     12  
Year                            
1992       89.59  89.59  89.59  
1993       84.15  84.15  84.15  
1994       79.47  79.47  79.47  
1995       74.58  74.58  74.58  
1996       69.60  69.60  69.60  
1997       65.73  65.73  65.73  
1998       63.20  63.20  63.20  
1999       59.70  59.70  59.70  
2000       54.49  54.49  54.49  
2001       50.24  50.24  50.24  
2002       47.89  47.89  47.89  
2003       44.61  44.61  44.61  
2004       40.84  40.84  40.84  
2005       36.22  36.22  36.22  
2006       31.96  31.96  31.96  
2007       28.30  28.30  28.30  
2008       23.56  23.56  23.56  
2009       24.00  24.00  24.00  
2010       22.00  22.00  22.00  
2011       18.27  18.27  18.27  
2012       15.87  15.87  15.87  
2013       14.19  14.19  14.19  
2014       12.37  12.37  12.37  
2015       12.24  12.24  12.24  
2016       10.84  10.84  10.84  
2017        8.53   8.53   8.53  
2018        5.94   5.94   5.94  
2019        4.06   4.06   4.06  
2020        2.79   2.79   2.79  
2021       -1.29  -1.29  -1.29